src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.cs

```diff
@@ -134,12 +134,31 @@ protected void SkipToAndParseCode(Func<HtmlSymbol, bool> condition)
                 {
                     if (last != null)
                     {
-                        Accept(last);
+                        if (!startOfLine || last.Type != HtmlSymbolType.WhiteSpace)
```

Comment:
Ahh. Instead of skipping the `last` we should conditionally accept it, mark its code generator as `null` (if there wasn't something prior to it like `<p>hi</p>   @* abcd *@`) to not mess up tooling.